chapter_optimization/sgd_vn.md

```diff
@@ -508,7 +512,14 @@ In particular, plot the distance from the optimal solution $(0, 0)$ as a functio
 5. Assume that $f(x) = x^2 (1 + \sin x)$. How many local minima does $f$ have? Can you change $f$ in such a way that to minimize it one needs to evaluate all local minima?
 -->
 
-*dịch đoạn phía trên*
+1. Hãy thử nghiệm với nhiều bộ định thời tốc độ học khác nhau trong SGD và với nhiều số vòng lặp khác nhau.
+Cụ thể, hãy vẽ biểu đồ khoảng cách từ nghiệm tối ưu $(0, 0)$ theo số vòng lặp.
+2. Chứng minh rằng với hàm $f(x_1, x_2) = x_1^2 + 2 x_2^2$, việc thêm nhiễu Gauss (*normal noise*) vào gradient tương đương với việc cực tiểu hoá hàm mất mát $l(\mathbf{x}, \mathbf{w}) = (x_1 - w_1)^2 + 2 (x_2 - w_2)^2$ trong đó $x$ tuân theo phân bố chuẩn.
+    * Xuất phát từ kì vọng và phương sai của phân phối cho $\mathbf{x}$.
+    * Chỉ ra rằng tính chất này nhìn chung có thể áp dụng cho hàm mục tiêu $f(\mathbf{x}) = \frac{1}{2} (\mathbf{x} - \mathbf{\mu})^\top Q (\mathbf{x} - \mathbf{\mu})$ for $Q \succeq 0$.
+3. So sánh sự hội tụ của SGD khi lấy mẫu từ $\{(x_1, y_1), \ldots, (x_m, y_m)\}$ không hoàn lại và khi lấy mẫu có hoàn lại.
+4. Bạn sẽ thay đổi chương trình SGD thế nào nếu như một số gradient (hoặc một số toạ độ liên kết với nó) liên tục lớn hơn so với tất cả các gradient khác?
+5. Giả sử rằng $f(x) = x^2 (1 + \sin x)$. $f$ có bao nhiêu cực tiểu? Bạn có thể thay đổi $f$ theo cách nào để giảm thiểu yêu cầu ước lượng tất cả các cực tiểu?
```

Comment:
```suggestion
5. Giả sử rằng $f(x) = x^2 (1 + \sin x)$. $f$ có bao nhiêu cực tiểu? Bạn có thể thay đổi $f$ theo cách nào để cực tiểu hóa nó, ta cần xét tất cả các cực tiểu?
```

Reply:
Theo em hiểu thì ý câu này nghĩa là `tối thiểu hoá nhu cầu phải ước lượng tất cả các cực tiểu` chứ không phải là `tối thiểu hoá hàm f`. Không biết em có hiểu đúng không :D

Reply:
Mình hiểu giống @rootonchair , mình để xuất sửa thêm cho đỡ rối 
```suggestion
5. Giả sử rằng $f(x) = x^2 (1 + \sin x)$. $f$ có bao nhiêu cực tiểu? Thay đổi hàm số sao cho để cực tiểu hóa giá trị hàm $f$, ta cần xét tất cả các điểm cực tiểu?
```

Reply:
@trGiang99 , câu gốc là:
`Can you change $f$ in such a way that to minimize it one needs to evaluate all local minima`

Thì mình hiểu ở đây `it` là hàm f còn `one` ở đây có nghĩa là `một người`. => `để cực tiểu hóa hàm f một người cần phải xét tất cả các cực tiểu`

Reply:
à em đọc nhầm thành `its one need` :( vậy em sửa lại theo ý của anh @lkhphuc nhé